### PR TITLE
[ETFE-2263] updated withAnswer to include optional redirectRoute parameter

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/controllers/sections/transportArranger/TransportArrangerNameController.scala [hunk 1]
```diff
@@ -67,22 +67,19 @@ class TransportArrangerNameController @Inject()(
...
   private def withTransportArrangerAnswer(f: TransportArranger => Future[Result])(implicit request: DataRequest[_]): Future[Result] = {
-    request.userAnswers.get(TransportArrangerPage) match {
-      case Some(transportArranger) if transportArranger == GoodsOwner | transportArranger == Other => f(transportArranger)
-      case Some(transportArranger) =>
+    withAnswer(
+      page = TransportArrangerPage,
+      // TODO: update redirectRoute to journey index page when built
+      redirectRoute = controllers.sections.transportArranger.routes.TransportArrangerController.onPageLoad(request.ern, request.lrn, NormalMode)
+    ) {
+      case transportArranger if transportArranger == GoodsOwner | transportArranger == Other => f(transportArranger)
```
Comment (LewisMitchell) [inline app/controllers/sections/transportArranger/TransportArrangerNameController.scala:75]: A minor improvement if you want to make it 
```suggestion
      case transportArranger@(GoodsOwner | Other) => f(transportArranger)
```

Comment (LewisMitchell) [reply app/controllers/sections/transportArranger/TransportArrangerNameController.scala:75]: (thanks Eric)

**Review by LewisMitchell — COMMENTED** [review]
Could you rebase also? The JT's currently fail

File: app/controllers/sections/destination/DestinationDetailsChoiceController.scala [hunk 2]
```diff
@@ -44,33 +44,29 @@ class DestinationDetailsChoiceController @Inject()(override val messagesApi: Mes
                                                   ) extends BaseNavigationController with AuthActionHelper {
 
   def onPageLoad(ern: String, lrn: String, mode: Mode): Action[AnyContent] =
-    authorisedDataRequest(ern, lrn) {
+    authorisedDataRequestAsync(ern, lrn) {
       implicit request =>
-        request.userAnswers.get(DestinationTypePage) match {
-          case Some(movementScenario) =>
-            Ok(view(
+        withAnswer(DestinationTypePage) {
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply app/controllers/sections/destination/DestinationDetailsChoiceController.scala:49]: In my opinion, no - it's one of the very first pages in the CaM journey, ahead of the LRN screen. If they haven't even hit DestinationTypePage yet, they won't be able to do the rest of the journey and might as well start again.

[the remark this answers is not in the document]
Comment (james-work-account) [reply app/controllers/sections/destination/DestinationDetailsChoiceController.scala:49]: If they're redirected to the DestinationTypePage, they might as well be starting again. We're intentionally not saving their user answers until after the INFO sub-section is complete for the first time, so if they don't have a DestinationTypePage, they haven't been through the INFO section for the first time, and are intentionally trying to get to a page which they shouldn't reach.

File: test/controllers/sections/transportUnit/TransportSealChoiceControllerSpec.scala [hunk 3]
```diff
@@ -197,7 +198,8 @@ class TransportSealChoiceControllerSpec extends SpecBase with MockUserAnswersSer
         val result = route(application, request).value
 
         status(result) mustEqual SEE_OTHER
-        redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+        redirectLocation(result).value mustEqual
+          controllers.sections.transportUnit.routes.TransportUnitIndexController.onPageLoad(testErn, testLrn).url
```
Comment (LewisMitchell) [inline test/controllers/sections/transportUnit/TransportSealChoiceControllerSpec.scala:202]: The test name says redirect to 'journey correction controller' I think this should be 'index controller'

File: test/controllers/sections/transportUnit/TransportSealChoiceControllerSpec.scala [hunk 4]
```diff
@@ -117,7 +117,8 @@ class TransportSealChoiceControllerSpec extends SpecBase with MockUserAnswersSer
...
```
Comment (LewisMitchell) [inline test/controllers/sections/transportUnit/TransportSealChoiceControllerSpec.scala:121]: The test name says redirect to 'journey correction controller' I think this should be 'index controller'

File: test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala [hunk 5]
```diff
@@ -153,7 +153,7 @@ class GuarantorVatControllerSpec extends SpecBase with MockUserAnswersService {
 
         status(result) mustEqual SEE_OTHER
         redirectLocation(result).value mustEqual
-          controllers.sections.guarantor.routes.GuarantorArrangerController.onPageLoad(testErn, testLrn, NormalMode).url
+          controllers.sections.guarantor.routes.GuarantorIndexController.onPageLoad(testErn, testLrn).url
```
Comment (LewisMitchell) [inline test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:156]: The test name says redirect to 'arranger controller' I think this should be 'index controller'

File: test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala [hunk 6]
```diff
@@ -170,7 +170,7 @@ class GuarantorVatControllerSpec extends SpecBase with MockUserAnswersService {
...
```
Comment (LewisMitchell) [inline test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:173]: The test name says redirect to 'arranger controller' I think this should be 'index controller'

File: test/controllers/sections/guarantor/GuarantorNameControllerSpec.scala [hunk 7]
```diff
@@ -152,7 +152,7 @@ class GuarantorNameControllerSpec extends SpecBase with MockUserAnswersService {
 
         status(result) mustEqual SEE_OTHER
         redirectLocation(result).value mustEqual
-          controllers.sections.guarantor.routes.GuarantorArrangerController.onPageLoad(testErn, testLrn, NormalMode).url
+          controllers.sections.guarantor.routes.GuarantorIndexController.onPageLoad(testErn, testLrn).url
```
Comment (LewisMitchell) [inline test/controllers/sections/guarantor/GuarantorNameControllerSpec.scala:155]: The test name says redirect to 'arranger controller' I think this should be 'index controller'

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: test/controllers/sections/guarantor/GuarantorNameControllerSpec.scala [hunk 8]
```diff
@@ -137,7 +137,7 @@ class GuarantorNameControllerSpec extends SpecBase with MockUserAnswersService {
         val result = route(application, request).value
 
         status(result) mustEqual SEE_OTHER
-        redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+        redirectLocation(result).value mustEqual controllers.routes.JourneyRecoveryController.onPageLoad().url
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply test/controllers/sections/guarantor/GuarantorNameControllerSpec.scala:140]: It does redirect to the journey recovery controller though? I think this test name is correct.

[the remark this answers is not in the document]
Comment (LewisMitchell) [reply test/controllers/sections/guarantor/GuarantorNameControllerSpec.scala:140]: Apologies, got carried away 🙈

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**